src/main/java/seedu/duke/models/logic/Api.java

```diff
@@ -0,0 +1,200 @@
+     *         the single prerequisite.
+     *         - If there are multiple prerequisites, it returns a JSON object representing the full prerequisite tree.
+     */
+    private static JSONObject getModulePrereqTree(String moduleCode) {
+        JSONObject fullModuleInfo = getFullModuleInfo(moduleCode);
+        if(fullModuleInfo == null){
+            return null;
+        }
+        //prereqTree can be returned as a string(single pre requisite), null(No pre requisites) or object
+        Object prereqTree = fullModuleInfo.get("prereqTree");
+        if(prereqTree == null){
+            return null;
+        }else if(prereqTree instanceof String){
+            JSONObject jsonObject = new JSONObject();
+            ArrayList<String> requirementList = new ArrayList<>();
+            requirementList.add((String) prereqTree);
+            jsonObject.put("or", requirementList);
+
+            return jsonObject;
+        }
+
+        return (JSONObject) fullModuleInfo.get("prereqTree");
+    }
+
+    private static boolean isModuleException(String moduleCode){
+        ArrayList<String> exemptedModules = new ArrayList<>();
+        exemptedModules.add("CS1231");
+        exemptedModules.add("MA1508E");
+        exemptedModules.add("EE4204");
+        return exemptedModules.contains(moduleCode);
+    }
+
+    private static ArrayList<String> getExemptedPrerequisite(String moduleCode){
+        HashMap<String, ArrayList<String>> map = new HashMap<>();
+        ArrayList<String> list1 = new ArrayList<>();
+        list1.add("MA1511");
+        list1.add("MA1512");
+        map.put("CS1231", list1);
+
+        ArrayList<String> list2 = new ArrayList<>();
+        list2.add("MA1511");
+        list2.add("MA1512");
+        map.put("MA1508E", list2);
+
+        ArrayList<String> list3 = new ArrayList<>();
+        list3.add("ST2334");
+        map.put("EE4204", list3);
+
+        return map.get(moduleCode);
+    }
+
+    public static JSONObject getFullModuleInfo(String moduleCode) {
+        try {
+            String url = "https://api.nusmods.com/v2/2023-2024/modules/" + moduleCode + ".json";
+            HttpClient client = HttpClient.newHttpClient();
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(new URI(url))
+                    .GET()
+                    .build();
+            HttpResponse<String> response = client.send(request, HttpResponse.BodyHandlers.ofString());
+            String responseBody = response.body();
+            JSONParser parser = new JSONParser();
+            // Will refactor the variable later on, left it for easier readability
+            return (JSONObject) parser.parse(responseBody);
+        } catch (ParseException e) {
+            //to be replaced with more robust error class in the future
+            System.out.println("Invalid Module Name");
+        } catch (IOException | InterruptedException e) {
+            System.out.println("Invalid Module Name");
```

Comment:
Consider using a standardized Ui object to print output onto command line. The ModulePlannerController class is using CommandLineView.